elasticapm/utils/__init__.py

```diff
@@ -27,23 +27,23 @@
 
 def varmap(func, var, context=None, name=None):
     """
-    Executes ``func(key_name, value)`` on all values
-    recurisively discovering dict and list scoped
+    Executes ``func(key_name, value)`` on all values,
+    recursively discovering dict and list scoped
     values.
     """
     if context is None:
-        context = {}
+        context = set()
     objid = id(var)
     if objid in context:
         return func(name, '<...>')
-    context[objid] = 1
+    context.add(objid)
     if isinstance(var, dict):
         ret = dict((k, varmap(func, v, context, k)) for k, v in compat.iteritems(var))
     elif isinstance(var, (list, tuple)):
-        ret = [varmap(func, f, context, name) for f in var]
+        ret = func(name, [varmap(func, f, context, name) for f in var])
     else:
         ret = func(name, var)
-    del context[objid]
+    context.remove(objid)
     return ret
```

Comment:
instead of the change on line 43 and instead of the `else` clause here, could this just be:

`return func(name, var)`

?

Reply:
that would run `func` on `dict` as well. Which isn't entirely wrong, but `shorten` doesn't have support for dicts yet. I'll try to add support for `dict` in `shorten` in a different PR and will change it to your suggested logic one that's in place